### DISCUSSION TITLE
Server error '500 ' for url 'https://askme.rblbank.com/callosum/v1/tspublic/v1/user/sync'

### DISCUSSION THREAD

**Discussion by GirishB859** [thoughtspot/cs_tools#153] [General]
Hi Team,

I fetched the user details using command "cs_tools tools user-management sync --syncer csv://directory=C:\Users\c35045\Documents\original_data\ --export" edited the fetched fill "ts_auth_sync_users" to add new users to the Thoughtspot then trying to sync back using command "cs_tools tools user-management sync --syncer csv://directory=C:\Users\c35045\Documents --new-user-password pass@321 --verbose" . And I am getting error "Server error '500 ' for url 'https://askme.rblbank.com/callosum/v1/tspublic/v1/user/sync' " Please let me know how to resolve it. Or how to check background service going on to resolve it,

**Comment by boonhapus** [comment]
Hey @GirishB859 ! To better troubleshoot, can you send the last few logs?

> 
> Run the following command.
> `cs_tools logs report --latest 5`
> Then use the secure file share link below, which routes to the CS Tools team directly. 
> https://thoughtspot.egnyte.com/ul/1bK0dz05L6

**Comment by GirishB859** [comment]
Hi @boonhapus ,

I have uploaded logs over the same link you shared,

**Comment by GirishB859** [comment]
Hi @boonhapus,

Any update?

**Reply by boonhapus** [reply]
Hey @GirishB859 , you're on an older version of CS Tools. Can you upgrade to the latest version? Please use this command..

`powershell -c "IRM https://raw.githubusercontent.com/thoughtspot/cs_tools/master/cs_tools/updater/_bootstrapper.py | python - --install"`

However what I did notice is this.. the error you're getting is due to conflicting usernames.

```
[TRACE - 2024-06-28 12:52:36,948] [cs_tools.api._client - _client.request 80] << CONTENT:

{"code":13004,"incident_id_guid":"7138088b-bf15-4af5-a128-fcfde83ec057","trace_id_guid":"7138088b-bf15-4af5-a128-fcfde83ec057","debug":"[\"Two users have the same name2775\",\"\"]"}
```

**Reply by GirishB859** [reply]
Hello @boonhapus 

I have installed cstools in the Python environment of Anaconda Navigator due to PowerShell being blocked in our office environment. Could you please provide me with the necessary command to update the tool in that specific environment?

**Reply by boonhapus** [reply]
I see @GirishB859. Do you remember if we installed with a zip file, or were we able to reach the Internet through conda?

**Comment by GirishB859** [comment]
Hi,

Yes we are able to reach internet using conda please share the command to
update cstools

**Reply by boonhapus** [reply]
@GirishB859 You'll need to activate your conda virtual environment and run this command

`python -m pip install "cs_tools[cli] @ https://github.com/thoughtspot/cs_tools/archive/v1.5.8.zip"`
